content/docs/manuals/extending-oakestra/hooks.md

```diff
@@ -0,0 +1,94 @@
+---
+title: "Setup Hooks"
+summary: ""
+draft: false
+weight: 4
+toc: true
+seo:
+  title: "" # custom title (optional)
+  description: "" # custom description (recommended)
+  canonical: "" # custom canonical URL (optional)
+  noindex: false # false (default) or true
+---
+
+## Introduction
+
+Hooks in Oakestra provide a mechanism for listening to and reacting to system life-cycle events. By leveraging hooks, developers can decouple components and enable services to dynamically respond to events like creation, updates, or deletion of system entities. This section explains how to use hooks effectively, with step-by-step instructions for registering and implementing them.
+
+Hooks allow developers to:
+1. **Subscribe to Events**: Register services to listen for system life-cycle events such as entity creation, updates, or deletion.
+2. **React to Events**: Perform specific actions when events occur, either synchronously (blocking) or asynchronously (non-blocking).
```

Comment:
Can you package this within a "tip callout" (https://getdoks.org/docs/basics/shortcodes/#custom-callout-titles)? You can use the title "Why should developers use hooks?"